bpf/lib/conntrack.h

```diff
@@ -230,14 +230,21 @@ static inline __u8 __inline__ __ct_lookup(void *map, struct __sk_buff *skb,
 		if (ct_state) {
 			ct_state->rev_nat_index = entry->rev_nat_index;
 			ct_state->loopback = entry->lb_loopback;
-			ct_state->slave = entry->slave;
 			ct_state->node_port = entry->node_port;
-			/* As we currently support both types of services (in the code
-			 * referred as "legacy" and "v2"), we store references to both
-			 * types of service endpoints. For the legacy, a slave slot
-			 * number is stored within the "ct_entry.slave" field, while
-			 * for the v2, a backend id is stored in the "ct_entry.rx_bytes"
-			 * field which is not used by a entry with dir=CT_SERVICE.
+			/* To support seamless upgrade from an earlier service
+			 * implementation, we store references to the backend
+			 * in two places in the entry: backend_id and rx_bytes.
+			 * Previously, the field "ct_entry.backend_id" was used
+			 * for legacy services so we co-opted the field
+			 * "ct_entry.rx_bytes" to store the services v2
+			 * backend (as it is not used with dir=CT_SERVICE).
+			 *
+			 * As of v1.6, "ct_entry.backend_id" is populated with
```

Comment:
I see. Do you think it would help to zero the field instead? Then we could change the ct_entry to:
```
...
union {
    __u32 backend_id;
    struct {
        __u16 unused;
        __u8 tx_flags_seen;
        __u8 rx_flags_seen;
    };
}
...
```

Reply:
^ This is what I've done in my latest push, let me know if you have other thoughts.

Reply:
Sorry for the confusion, what I meant is that I changed it to zero the field.

I haven't modified the structure of ct_entry yet as I wasn't sure how it would impact the alignchecker.